src/lib.rs

```diff
@@ -68,24 +68,23 @@ impl Correctness {
         assert_eq!(answer.len(), 5);
         assert_eq!(guess.len(), 5);
         let mut c = [Correctness::Wrong; 5];
-        let mut used = [false; 5];
-        // Mark things green
-        for (i, (a, g)) in answer.bytes().zip(guess.bytes()).enumerate() {
-            if a == g {
+        let answer_bytes = answer.as_bytes();
+        let guess_bytes = guess.as_bytes();
+
+        for (i, &a) in answer_bytes.iter().enumerate() {
+            if a == guess_bytes[i] {
+                // Mark things green
                 c[i] = Correctness::Correct;
-                used[i] = true;
+            } else if let Some(j) = guess_bytes
```

Comment:
I think a comment here would be really useful. I _think_ what this is doing is looking for the first occurrence of a given answer letter in the guess that's a) not already correct and b) not already marked as misplaced, but parsing that took me a while. I think I would actually expand out that closure with `{}` and introduce local variables (which will get optimized out anyway) to denote what each comparison checks for.

Reply:
I hope the comment, I added, does explain it a little bit better 😄 I didn't really know how to name the extracted variables, hope the comment is ok.